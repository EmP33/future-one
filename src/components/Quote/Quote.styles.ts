import styled from "styled-components";

export const Wrapper = styled.section`
  display: grid;
  grid-template-columns: 0.75fr 1.5fr;
  column-gap: 4rem;

  @media only screen and (max-width: 1200px) {
    column-gap: 2rem;
  }

  @media only screen and (max-width: 900px) {
    grid-template-columns: 1fr 1.5fr;
    column-gap: 0;
  }

  @media only screen and (max-width: 600px) {
    grid-template-columns: 1fr;
  }

  .quote__heading {
    border-left: 10px solid var(--color-green);
    color: var(--color-green);
    padding: 5rem 2.5rem 5rem 1.75rem;
    margin: 1.5rem 0;

    @media only screen and (max-width: 1200px) {
      padding: 3rem 1.25rem 3rem;
    }

    @media only screen and (max-width: 900px) {
      padding: 2.5rem 1rem 2.5rem;
    }
  }

  .quote-content {
    display: grid;
    align-content: center;
    width: 75%;
    row-gap: 0.3rem;

    @media only screen and (max-width: 900px) {
      width: 100%;
    }

    & blockquote {
      font-style: italic;
    }

    &__footer {
      display: flex;
      align-items: center;
      justify-content: space-between;
    }

    .quoteIcon--1 {
      margin-bottom: 0.5rem;
    }

    .quoteIcon--2 {
      transform: rotate(180deg);
    }
  }
`;
