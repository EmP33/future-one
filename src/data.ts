export interface IQuestion {
  id: string;
  question: string;
  answer: string;
}

export interface IQuote {
  id: string;
  title: string;
  content: string;
  quoteAuthor: string;
}

export interface ICategory {
  id: number;
  iconName: string;
}

export const questions: IQuestion[] = [
  {
    id: "01",
    question: "Arcu non odio",
    answer:
      "Sapien nec sagittis aliquam malesuada bibendum. Tempor nec feugiat nisl pretium.",
  },
  {
    id: "02",
    question: "Eget gravida cum sociis",
    answer:
      "Sapien nec sagittis aliquam malesuada bibendum. Tempor nec feugiat nisl pretium.",
  },
  {
    id: "03",
    question: "Massa massa ultricies ",
    answer:
      "Sapien nec sagittis aliquam malesuada bibendum. Tempor nec feugiat nisl pretium.",
  },
];

export const quote: IQuote = {
  id: "q1",
  title: "Bibendum at varius vel pharetra vel turpis nunc eget lorem.",
  content:
    "Aliquam purus sit amet luctus venenatis lectus magna. Faucibus purus in massa tempor nec feugiat nisl pretium fusce. Tortor vitae purus faucibus ornare. Neque viverra justo nec ultrices. Nisl nunc mi ipsum faucibus. Tristique risus nec feugiat in fermentum. Elit duis tristique sollicitudin nibh sit amet commodo nulla.",
  quoteAuthor: "Faucibus Vitae, Office Assistant",
};

export const categories: ICategory[] = [
  { id: 0, iconName: "Lifebuoy" },
  { id: 1, iconName: "Scissors" },
  { id: 2, iconName: "Android" },
  { id: 3, iconName: "Zoom" },
];
