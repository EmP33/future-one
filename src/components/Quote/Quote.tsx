import React from "react";
import { Wrapper } from "./Quote.styles";
import { logo, quote as quoteIcon } from "../../assets";
import { IQuote } from "../../data";

const Quote: React.FC<{ quote: IQuote }> = ({ quote }) => {
  return (
    <Wrapper data-aos="fade-right">
      <div className="quote__heading">
        <h2>{quote.title}</h2>
      </div>
      <div className="quote-content">
        <div className="quoteIcon quoteIcon--1">
          <img src={quoteIcon} alt="quote" width={32} height={32} />
        </div>
        <blockquote>{quote.content}</blockquote>
        <div className="quote-content__footer">
          <p>{quote.quoteAuthor}</p>
          <div className="quoteIcon quoteIcon--2">
            <img src={quoteIcon} alt="quote" width={32} height={32} />
          </div>
        </div>
        <div className="logo">
          <img src={logo} alt="invision" width={86} height={29} />
        </div>
      </div>
    </Wrapper>
  );
};

export default Quote;
