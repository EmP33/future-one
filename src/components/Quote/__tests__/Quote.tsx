import Quote from "../Quote";
import ShallowRenderer from "react-test-renderer/shallow";
import { quote } from "../../../data";

describe("<Quote />", () => {
  it("should render properly", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Quote quote={quote} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
