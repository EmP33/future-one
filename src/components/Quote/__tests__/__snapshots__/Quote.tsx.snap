// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quote /> should render properly 1`] = `
<styled.section
  data-aos="fade-right"
>
  <div
    className="quote__heading"
  >
    <h2>
      Bibendum at varius vel pharetra vel turpis nunc eget lorem.
    </h2>
  </div>
  <div
    className="quote-content"
  >
    <div
      className="quoteIcon quoteIcon--1"
    >
      <img
        alt="quote"
        height={32}
        src="quote.webp"
        width={32}
      />
    </div>
    <blockquote>
      Aliquam purus sit amet luctus venenatis lectus magna. Faucibus purus in massa tempor nec feugiat nisl pretium fusce. Tortor vitae purus faucibus ornare. Neque viverra justo nec ultrices. Nisl nunc mi ipsum faucibus. Tristique risus nec feugiat in fermentum. Elit duis tristique sollicitudin nibh sit amet commodo nulla.
    </blockquote>
    <div
      className="quote-content__footer"
    >
      <p>
        Faucibus Vitae, Office Assistant
      </p>
      <div
        className="quoteIcon quoteIcon--2"
      >
        <img
          alt="quote"
          height={32}
          src="quote.webp"
          width={32}
        />
      </div>
    </div>
    <div
      className="logo"
    >
      <img
        alt="invision"
        height={29}
        src="logo.webp"
        width={86}
      />
    </div>
  </div>
</styled.section>
`;
